File: Python/underautomation/universal_robots/rtde/rtde_inputs_description.py
import typing
from underautomation.universal_robots.rtde.rtde_input_data_description import RtdeInputDataDescription
from underautomation.universal_robots.rtde.rtde_input_data import RtdeInputData
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Rtde import RtdeInputsDescription as rtde_inputs_description

class RtdeInputsDescription:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = rtde_inputs_description()
		else:
			self._instance = _internal
	def get(self, data: RtdeInputData) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.Get(data._instance))
	@property
	def items(self) -> typing.Any:
		return self._instance.Items
	@property
	def speed_slider_mask(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.SpeedSliderMask)
	@property
	def speed_slider_fraction(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.SpeedSliderFraction)
	@property
	def standard_digital_output_mask(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.StandardDigitalOutputMask)
	@property
	def configurable_digital_output_mask(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.ConfigurableDigitalOutputMask)
	@property
	def standard_digital_output(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.StandardDigitalOutput)
	@property
	def configurable_digital_output(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.ConfigurableDigitalOutput)
	@property
	def standard_analog_output_mask(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.StandardAnalogOutputMask)
	@property
	def standard_analog_output_type(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.StandardAnalogOutputType)
	@property
	def standard_analog_output0(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.StandardAnalogOutput0)
	@property
	def standard_analog_output1(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.StandardAnalogOutput1)
	@property
	def input_bt_registers0_to31(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.InputBtRegisters0To31)
	@property
	def input_bt_registers32_to63(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.InputBtRegisters32To63)
	@property
	def input_bit_registers(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.InputBitRegisters)
	@property
	def input_int_registers(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.InputIntRegisters)
	@property
	def input_double_registers(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.InputDoubleRegisters)
	@property
	def external_force_torque(self) -> RtdeInputDataDescription:
		return RtdeInputDataDescription(self._instance.ExternalForceTorque)

File: UnderAutomation.UniversalRobots.ObfuscatedSources/Rtde/RtdeInputsDescription.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using System.Collections.ObjectModel;

namespace Rtde {

	public class RtdeInputsDescription {


		public RtdeInputDataDescription Get(RtdeInputData data)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public ReadOnlyCollection<RtdeInputDataDescription> Items { get; }

		/// <summary>
		/// 0 = don't change speed slider with this input, 1 = use speed_slider_fraction to set speed slider value
		/// </summary>
		public RtdeInputDataDescription SpeedSliderMask { get; }

		/// <summary>
		/// new speed slider value
		/// </summary>
		public RtdeInputDataDescription SpeedSliderFraction { get; }

		/// <summary>
		/// Standard digital output bit mask
		/// </summary>
		public RtdeInputDataDescription StandardDigitalOutputMask { get; }

		/// <summary>
		/// Configurable digital output bit mask
		/// </summary>
		public RtdeInputDataDescription ConfigurableDigitalOutputMask { get; }

		/// <summary>
		/// Standard digital outputs
		/// </summary>
		public RtdeInputDataDescription StandardDigitalOutput { get; }

		/// <summary>
		/// Configurable digital outputs
		/// </summary>
		public RtdeInputDataDescription ConfigurableDigitalOutput { get; }

		/// <summary>
		/// Standard analog output mask
		/// </summary>
		public RtdeInputDataDescription StandardAnalogOutputMask { get; }

		/// <summary>
		/// Output domain {0=current[mA], 1=voltage[V]}. Bits 0-1: standard_analog_output_0 | standard_analog_output_1
		/// </summary>
		public RtdeInputDataDescription StandardAnalogOutputType { get; }

		/// <summary>
		/// Standard analog output 0 (ratio) [0..1]
		/// </summary>
		public RtdeInputDataDescription StandardAnalogOutput0 { get; }

		/// <summary>
		/// Standard analog output 1 (ratio) [0..1]
		/// </summary>
		public RtdeInputDataDescription StandardAnalogOutput1 { get; }

		/// <summary>
		/// General purpose bits. This range of the boolean input registers is reserved for FieldBus/PLC interface usage.
		/// </summary>
		public RtdeInputDataDescription InputBtRegisters0To31 { get; }

		/// <summary>
		/// General purpose bits. This range of the boolean input registers is reserved for FieldBus/PLC interface usage.
		/// </summary>
		public RtdeInputDataDescription InputBtRegisters32To63 { get; }

		/// <summary>
		/// 64 general purpose bits. X: [64..127] - The upper range of the boolean input registers can be used by external RTDE clients (i.e URCAPS).
		/// </summary>
		public RtdeInputDataDescription InputBitRegisters { get; }

		/// <summary>
		/// 48 general purpose integer registers. X: [0..23] - The lower range of the integer input registers is reserved for FieldBus/PLC interface usage. X: [24..47] - The upper range of the integer input registers can be used by external RTDE clients (i.e URCAPS).
		/// </summary>
		public RtdeInputDataDescription InputIntRegisters { get; }

		/// <summary>
		/// 48 general purpose double registers. X: [0..23] - The lower range of the double input registers is reserved for FieldBus/PLC interface usage. X: [24..47] - The upper range of the double input registers can be used by external RTDE clients (i.e URCAPS).
		/// </summary>
		public RtdeInputDataDescription InputDoubleRegisters { get; }

		/// <summary>
		/// Input external wrench when using ft_rtde_input_enable builtin.
		/// </summary>
		public RtdeInputDataDescription ExternalForceTorque { get; }
	}
}
